util/test/chpl_launchcmd.py

```diff
@@ -235,24 +235,13 @@ def run(self):
             job_id = self.submit_job(testing_dir, output_file, input_file)
             logging.info('Test has been queued (job id: {0}). Waiting for output...'.format(job_id))
 
-            # TODO: The while condition here should look for jobs that become held,
-            #       are in the queue too long, or ??? and do something
-            #       intelligent. For example, if the job is in the queue longer
-            #       than the walltime, it should probably be deleted (qdel
-            #       <job_id>) and a timeout should be reported. Here are all the
-            #       pbs (torque) job statuses:
-            #
-            #           C -     Job is completed after having run/
-            #           E -  Job is exiting after having run.
-            #           H -  Job is held.
-            #           Q -  job is queued, eligible to run or routed.
-            #           R -  job is running.
-            #           T -  job is being moved to new location.
-            #           W -  job is waiting for its execution time
-            #                (-a option) to be reached.
-            #           S -  (Unicos only) job is suspend.
-            #
-            #       (thomasvandoren, 2014-04-09)
+            def wait_for_file(filename):
+                """Wait for filename for a few seconds. If it shows up, return early."""
+                for i in xrange(10):
+                    if os.path.exists(filename):
+                        break
+                    else:
+                        time.sleep(0.5)
```

Comment:
Is there a version of sleep that returns the amount of time slept for?

If so I think this would look cleaner as: 

```python
outputWaitTime = 0.0
while (outputWaitTime < 5.0 and not os.path.exists(filename):
    outputWaitTime += time.sleep(0.5)
```